pennylane/qchem/vibrational/pes_generator.py

```diff
@@ -0,0 +1,676 @@
+    local_harmonic_pes = np.zeros_like(local_pes_onebody)
+    if do_dipole:
+        local_dipole_onebody = np.zeros((nmodes, len(jobs_on_rank), 3), dtype=float)
+        ref_dipole = get_dipole(scf_result, method)
+    for mode in range(nmodes):
+        displ_vec = displ_vecs[mode]
+        if (freqs[mode].imag) > 1e-6:
+            continue # pragma: no cover
+
+        job_idx = 0
+        for job in jobs_on_rank:
+            pt = gauss_grid[job]
+            # numerical scaling out front to shrink region
+            scaling = np.sqrt(HBAR / (2 * np.pi * freqs[mode] * 100 * C_LIGHT))
+            positions = np.array(init_geom + scaling * pt * displ_vec)
+
+            displ_mol = qml.qchem.Molecule(
+                molecule.symbols,
+                positions,
+                basis_name=molecule.basis_name,
+                charge=molecule.charge,
+                mult=molecule.mult,
+                unit="angstrom",
+                load_data=True,
+            )
+
+            displ_scf = single_point(displ_mol, method=method)
+
+            omega = freqs_au[mode]
+            ho_const = omega / 2
+
+            local_harmonic_pes[mode][job_idx] = ho_const * (pt**2)
+
+            local_pes_onebody[mode][job_idx] = displ_scf.e_tot - scf_result.e_tot
+            if do_dipole:
+                local_dipole_onebody[mode, job_idx, :] = get_dipole(displ_scf, method) - ref_dipole
+            job_idx += 1
+
+    if do_dipole:
+        return local_pes_onebody, local_dipole_onebody
+    return local_pes_onebody, None
+
+
+def _load_pes_onemode(num_pieces, nmodes, quad_order, do_dipole=False):
+    """
+    Loader to combine pes_onebody and dipole_onebody from multiple processors.
+    """
+
+    pes_onebody = np.zeros((nmodes, quad_order), dtype=float)
+
+    if do_dipole:
+        dipole_onebody = np.zeros((nmodes, quad_order, 3), dtype=float)
+
+    for mode in range(nmodes):
+        init_chunk = 0
+        for piece in range(num_pieces):
+            f = h5py.File("v1data" + f"_{piece}" + ".hdf5", "r+")
+            local_pes_onebody = f["V1_PES"][()]
+            if do_dipole:
+                local_dipole_onebody = f["D1_DMS"][()]
+
+            end_chunk = np.array(local_pes_onebody).shape[1]
+            pes_onebody[mode][init_chunk : init_chunk + end_chunk] = local_pes_onebody[mode]
+            if do_dipole:
+                dipole_onebody[mode][init_chunk : init_chunk + end_chunk] = local_dipole_onebody[
+                    mode
+                ]
+            init_chunk += end_chunk
+
+    if do_dipole:
+        return pes_onebody, dipole_onebody
+    return pes_onebody, None
+
+
+def pes_twomode(
+    molecule,
+    scf_result,
+    freqs_au,
+    displ_vecs,
+    gauss_grid,
+    pes_onebody,
+    dipole_onebody,
+    method="rhf",
+    do_dipole=False,
+):
+    r"""Computes the two-mode potential energy surface on a grid in real space, along the normal coordinate directions
+    (or any directions set by the displ_vecs).
```

Comment:
Not sure I understand this. It is alomng the normal coordinates or directions set by the displ_vecs? Are they the same?